File: src/etch/bindings/java/util/test/TestDateSerializer.java
package etch.bindings.java.util.test;


import static org.junit.Assert.assertEquals;

import java.util.Date;

import org.junit.Test;

import etch.bindings.java.msg.ImportExportHelper;
import etch.bindings.java.msg.StructValue;
import etch.bindings.java.msg.Type;
import etch.bindings.java.support.Class2TypeMap;
import etch.bindings.java.util.DateSerializer;

/** Tests DateSerializer */
public class TestDateSerializer
{
	/** @throws Exception */
	@Test
	public void testImport() throws Exception
	{
		Date date = new Date();
		testDate( date );
	}
	
	private void testDate( Date date )
	{
//		System.out.println( "---------------------------------------------" );
//		System.out.println( "date = "+date );
		
		Type type = new Type( "date" );
		Class2TypeMap class2type = new Class2TypeMap();
		DateSerializer.init( type, class2type );
		ImportExportHelper helper = type.getImportExportHelper();
		
		StructValue sv = helper.exportValue( date );
//		System.out.println( "sv = "+sv );
		
		assertEquals( sv.type(), type );
		
		Date date2 = (Date) helper.importValue( sv );
//		System.out.println( "date2 = "+date2 );
		
		assertEquals( date, date2 );
	}
}

File: src/etch/bindings/csharp/util/test/TestDateSerializer.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using Etch.Msg;
using Etch.Support;

namespace Etch.Util.Test
{
    [TestFixture]
    public class TestDateSerializer
    {

        [TestFixtureSetUp]
        public void First()
        {
            Console.WriteLine();
            Console.Write( "TestDateSerializer" );
        }


        [Test]
        public void TestImport()
        {
            DateTime date = new DateTime();
            TestDate( date );
        }

        private void TestDate( DateTime date )
        {
            XType type = new XType( "date" );
          
            Class2TypeMap class2type = 
                new Class2TypeMap();
            
            DateSerializer.Init( type, class2type );

            ImportExportHelper helper = type.GetImportExportHelper();
            StructValue sv = helper.ExportValue( date );

            Assert.AreEqual( sv.GetXType, type );

            DateTime date2 = ( DateTime ) helper.ImportValue( sv );

            Assert.AreEqual( date, date2 );
        }
    }
}
